### DISCUSSION TITLE
Does systray support child-menus

### DISCUSSION THREAD

**Discussion by johnsiilver** [fyne-io/fyne#3482] [Q&A] [answered]
Hey everyone,

I know the systray support is brand new.  I was wondering a couple things:
 - Does the menu items support child menus?  
 - Is it possible to launch the systray without a visible windows?
I mainly want a systray to launch with no windows on start.  But I wasn't able to figure out how to get that to happen.
The second thing is I *believe* I added child menus correctly, but can't see them.  I'm guessing that maybe this isn't supported in the current release.  

Sorry to ask on here, but the only documentation I could find on systray was the example.  I'd be happy to go read more documentation on just systray if someone has a pointer.

Thanks for any help and thanks for the cool package.

**Comment by andydotxyz** [comment, marked as answer]
Yes submenus are supported, in the nearly-released v2.3.0 (you can use develop branch).

You can show systray with no windows - just don't show any windows when you call `Run()` (instead of `ShowAndRun`)

**Comment by johnsiilver** [comment]
Thank you Andy for the answer.  And thank you for your time, I know it
takes a lot of energy to answer questions on open source software and
putting something this difficult out into the world (especially when it's
free).

Cheers!



On Mon, Dec 19, 2022 at 2:36 PM Andy Williams ***@***.***>
wrote:

> Yes submenus are supported, in the nearly-released v2.3.0 (you can use
> develop branch).
>
> You can show systray with no windows - just don't show any windows when
> you call Run() (instead of ShowAndRun)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/fyne-io/fyne/discussions/3482#discussioncomment-4452816>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAJQZLQSWAIMGAIE6IFDLK3WODPNZANCNFSM6AAAAAATDUAHDA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
John Doak
www.obscuredworld.com

**Comment by andydotxyz** [comment]
Thanks, you are very welcome :)
